tests/unit/test_utils.py

```diff
@@ -1,23 +1,46 @@
 # coding: utf-8
-from hapic.util import LowercaseDictKeys
+import pytest
+
+from hapic.exception import HapicException
+from hapic.util import LowercaseKeysDict
 
 
 class TestUtils(object):
     def test_unit__get__ok__nominal_case(self):
-        lowercase_dict = LowercaseDictKeys([("foo", "bar")])
+        lowercase_dict = LowercaseKeysDict([("foo", "bar")])
         assert "bar" == lowercase_dict.get("foo")
         assert "bar" == lowercase_dict.get("FOO")
 
     def test_unit__by_key__ok__nominal_case(self):
-        lowercase_dict = LowercaseDictKeys([("foo", "bar")])
+        lowercase_dict = LowercaseKeysDict([("foo", "bar")])
         assert "bar" == lowercase_dict["foo"]
         assert "bar" == lowercase_dict["FOO"]
 
     def test_unit__in__ok__nominal_case(self):
-        lowercase_dict = LowercaseDictKeys([("foo", "bar")])
+        lowercase_dict = LowercaseKeysDict([("foo", "bar")])
         assert "foo" in lowercase_dict
         assert "FOO" in lowercase_dict
 
     def test_unit__del__ok__nominal_case(self):
-        lowercase_dict = LowercaseDictKeys([("foo", "bar")])
+        lowercase_dict = LowercaseKeysDict([("foo", "bar")])
         del lowercase_dict["FOO"]
+
+    def test_unit__create__ok__nominal_case(self):
+        lowercase_dict = LowercaseKeysDict([("foo", "bar")])
+        assert {"foo": "bar"} == lowercase_dict
+
+    def test_unit__create__error__refuse_key(self):
+        with pytest.raises(HapicException):
+            LowercaseKeysDict([("FOO", "bar")])
+
+    def test_unit__set__error__refuse_key(self):
+        lowercase_dict = LowercaseKeysDict()
+
+        with pytest.raises(HapicException):
+            lowercase_dict.setdefault("FOO", "bar")
+
+    def test_unit__update__error__refuse_key(self):
+        lowercase_dict = LowercaseKeysDict()
+
+        with pytest.raises(HapicException):
```

Comment:
Exception should be specific for a specific case, no?

Reply:
Changed.